clang/test/CXX/drs/cwg27xx.cpp

```diff
@@ -18,6 +30,38 @@ void f(B b) {
 struct D : B {};
 } // namespace cwg2718
 
+namespace cwg2749 { // cwg2749: 19
+
+extern int x[2];
+struct Y {
+  int i;
+  int j;
+};
+extern Y y[2];
+
+#if __cplusplus >= 201103L
+static_assert(static_cast<void*>(x + 0) < static_cast<void*>(x + 1), "");
+static_assert(static_cast<void*>(&y[0].i) < static_cast<void*>(&y[0].j), "");
+static_assert(static_cast<void*>(&y[0].j) < static_cast<void*>(&y[1].i), "");
+#else
+enum X {
+  a = static_cast<void*>(x + 0) < static_cast<void*>(x + 1),
```

Comment:
You should copy over the C++98 "backport" of `static_assert` from e.g. (we use it liberally across DR test suite)
https://github.com/llvm/llvm-project/blob/0eb9e021b15f477b796a6aaed94c94887f19171c/clang/test/CXX/drs/cwg1xx.cpp#L8-L11

Reply:
C++98 has a separate test because `_Static_assert` doesn't constant fold its arguments in C++ (only in C as an extension), and in C++98 §6.4 "Constant expressions" "an *integral constant expression* shall have integral type and shall only have operands that are integer constants, enumeration constants, character constants, `sizeof` expressions, and floating constants that are the immediate operands of cast", causing `static_assert((x + 0) < (x + 1))` to fail because it has address operands.

On second thought, this change only applies in C++98 when we do constant folding as an extension. I don't think we need to test that a now-non-existent warning isn't issued in C++98 mode when we use an extension, so I'll just get rid of the C++98 test

Reply:
Feel free to remove the test in C++98 if it doesn't fit the requirements of integral constant expression, but for the future reference we also use the following in some DR tests: https://github.com/llvm/llvm-project/blob/0eb9e021b15f477b796a6aaed94c94887f19171c/clang/test/CXX/drs/cwg1xx.cpp#L13-L17